openehr-rm/src/main/java/com/nedap/archie/rminfo/UpdatedValueHandler.java

```diff
@@ -35,6 +40,44 @@
         if(parent instanceof CodePhrase) {
             return fixCodePhrase(rmObject, archetype, pathOfParent);
         }
+        if (parent instanceof DvQuantity) {
+            return fixDvQuantity(rmObject, archetype, pathOfParent);
+        }
+
+        return new HashMap<>();
+    }
+
+    private static Map<String, Object> fixDvQuantity(Object rmObject, Archetype archetype, String pathOfParent) {
+        try {
+            Map<String, Object> result = new HashMap<>();
+
+            RMPathQuery rmPathQuery = new RMPathQuery(pathOfParent);
+            DvQuantity quantity = rmPathQuery.find(ArchieRMInfoLookup.getInstance(), rmObject);
+
+            OperationalTemplate template = (OperationalTemplate) archetype;
+
+            CAttribute units = template.getDefinition().itemAtPath(pathOfParent + "/units");
+            CAttribute precision = template.getDefinition().itemAtPath(pathOfParent + "/precision");
+
+            if (units.getChildren().size() != 1 || precision.getChildren().size() != 1)
+                return result; // Only fix if there is 1 unit and 1 precision, don't assume anything otherwise
```

Comment:
You could split these two checks for more auto-fixing.

Also note that precision.getChildren() contains intervals, and you cannot use those.

Reply:
>and you cannot use those.

Not exactly sure what you mean, am I not allowed to do a `precision.getChildren().size() != 1`?